docs/design/request-authentication.md

```diff
@@ -0,0 +1,384 @@
+  applies. The requirement specifies which JWT providers should be used. This field is built from a HTTPRoute
+  `matches` rule that references the Authentication extended filter. When a referenced Authentication specifies
+  multiple `jwtProviders`, the JWT is considered valid if __any__ of the providers successfully validate the JWT.
+
+The following JWT Authentication filter `providers` configuration is created from the above Authentication.
+
+```yaml
+providers:
+   example:
+     issuer: https://www.example.com
+     audiences:
+     - foo.com
+     remote_jwks:
+       http_uri:
+         uri: https://foo.com/jwt/public-key/jwks.json
+         cluster: example_jwks_cluster
+         timeout: 1s
+```
+
+The following JWT Authentication filter `rules` configuration is created from the above HTTPRoute.
+
+```yaml
+rules:
+  - match:
+      prefix: /foo
+    requires:
+      provider_name: example
+```
+
+If the Authentication included a second JWT provider named `example`, the resulting JWT Authentication filter `rules`
+would be created:
+
+```yaml
+rules:
+- match:
+    prefix: /foo
+  requires:
+    requires_any:
+      requirements:
+      - provider_name: example
+      - provider_name: example2
+```
+
+### Implementation Outline
+
+* Create a `security` API group and add the Authentication API type to this group.
+* Update the Kubernetes provider to get/watch Authentication resources that are referenced by managed HTTPRoutes. Add
+  the referenced Authentication object to the resource map and publish it.
+* Update the resource translator to include the Authentication API in HTTPRoute processing.
+* Update the xDS translator to translate an Authentication into xDS resources. The translator should perform the
+  following:
+  * Convert a list of JWT rules from the xds IR into an Envoy JWT filter config.
+  * Create a JWT authentication filter.
+  * Build the HTTP Connection Manager (MCM) HTTP filters.
+  * Build the HCM.
+  * When building the Listener, create an HCM for each filter-chain.
+
+## Adding Authentication Provider Types
+
+Additional authentication provider types can be added in the future through the `ProviderType` API. For example, to add
+the `Foo` authentication provider type:
+
+Define the `FooProvider` type:
+
+```go
+package v1alpha1
+
+// FooProvider defines the "Foo" authentication provider type.
+type FooProvider struct {
+	// TODO: Define fields of the Foo authentication provider type.
+}
+```
+
+Add the `FooProvider` type to `AuthenticationSpec`:
+
+```go
+package v1alpha1
+
+type AuthenticationSpec struct {
+	...
+	
+	// Foo defines the Foo authentication type. For additional
+	// details, see:
+	//
+	//   <INSERT_LINK>
+	//
+	// +optional
+	Foo *FooAuthentication `json:"foo"`
+}
+```
+
+Authentication should support additional authentication types in the future, for example:
+- mutualTLS (client certificate)
```

Comment:
does this mean mutual TLS with the app backend or client mutual TLS with the Envoy proxy?

tbh regardless, should we exclude this one from this level of configuration for now, since it is a connection level mechanism? This seems like something for the former that we will be able to use the "Backend Properties" GEP to do and for the latter something we need to configure on a Gateway Listener

Reply:
This is to support client certificate to Envoy Gateway, so nothing to do with gateway to app backend (i.e. "Backend Properties").

This statements is actually meant to exclude mTLS from this design so deferred to future.